src/analyses/callstringAnalysis.ml

```diff
@@ -0,0 +1,87 @@
+open Analyses
+open GoblintCil
+open GobConfig
+
+(* Specifies the type of the callstack elements for the callstring analysis*)
+module type Callstack_Type =
+sig
+  include CilType.S
+  val stackTypeName: string
+  val pushElem: fundec -> ('d,'g,'c,'v) ctx -> t option (* returns an elements that should be pushed to the Callstack *)
+end
+
+(** Analysis with the k-callsting approach, which uses the last k callstack elements as context.
+    With the CT argument it is possible to specify the type of the callstack elements*)
+module Spec (CT:Callstack_Type) : MCPSpec = 
+struct
+  include Analyses.IdentitySpec
+
+  (* simulates a call stack of depth k*)
+  module CallStack = struct
+    include Printable.PQueue (CT)
+    let depth = get_int "ana.context.callStack_height" (* must be >= 0 *)
+
+    let push stack elem = (* pushes elem to the stack, guarantees stack depth of k*)
+      match elem with
+      | None -> stack
+      | Some e -> 
+        let new_stack = BatDeque.cons e stack in (* pushes new element to stack*)         
+        (* removes element from stack, if stack was filled with k elements*)
+        match (BatDeque.size new_stack - depth) with
+        | x when x <= 0 -> new_stack
+        | 1 -> fst @@ Option.get (BatDeque.rear new_stack)
+        | _ -> failwith "Callstack Error: It shouldn't happen that more than one element must be deleted to maintain the correct height!"
+  end
+
+  module D = Lattice.Flat (CallStack)
```

Comment:
We're being a bit cheeky here by abusing local states to in fact do contexts (and we're forced into it by the framework in sense). 
We probably should add a comment here what we are doing so people not deep into Goblint have an idea what's happening.